x-pack/plugins/fleet/public/applications/fleet/app.tsx

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useEffect, useState } from 'react';
+import { AppMountParameters } from 'kibana/public';
+import { EuiCode, EuiEmptyPrompt, EuiErrorBoundary, EuiPanel } from '@elastic/eui';
+import { createHashHistory, History } from 'history';
+import { Router, Redirect, Route, Switch } from 'react-router-dom';
```

Comment:
Note the subtle  change from using `HashRouter` to using `Router` along with History's `createHashHistory`. This is so that it can support testing.